### ISSUE TITLE
Cant connect to KeepassXC database

### ISSUE THREAD

**Issue by Romich777** [keepassxreboot/keepassxc-browser#1671]
Ubuntu - 22.04
Firefox - 102,  installed from PPA **not from Snap**
Keepassxc-browser - 1.8.0

Getting **Key exchange was not successful.** when trying to reload connection

Extension has to search native-messaging-hosts/org.keepassxc.keepassxc_browser.json in different locations for deb, snap and flatpack instals

**Comment by varjolintu** [comment]
Go through https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide and report your findings. Also, search for other threads with similar problems.

**Comment by dark-rider** [comment]
I suddenly have the same problem, since today, with Firefox 102.0.1 (yesterday with 102.0 it still worked, although since build 1.8.0, auto-connect does not work any more and you now have to connect manually, which now also does not work any more).

**Comment by varjolintu** [comment]
Auto-Reconnect was disabled for 1.8.0 because it has some unsolved memory leaks. Manual reconnect is mandatory until we get the issue solved.

**Comment by dark-rider** [comment]
Thank you for your quick answer. But as mentioned, now even manual reconnect does not work any more.

**Comment by varjolintu** [comment]
@dark-rider Well, there's no changes in the extension. Maybe there's a bug in Firefox update. Are you using Linux? I've seen a couple of reports before that some distros have failed packaging a new version and it can cause some strange problems.

**Comment by dark-rider** [comment]
Now, manual reconnect is working again, without any changes in the meantime.

**Comment by unicorn855** [comment]
> Auto-Reconnect was disabled for 1.8.0 because it has some unsolved memory leaks. Manual reconnect is mandatory until we get the issue solved.

Thank you @varjolintu for pointing that out, I was wondering where that experimental setting went and thought I overlooked something :-) Do you need any further information in order to debug this, or is there already a roadmap to solving this?

**Comment by varjolintu** [comment]
@unicorn855 Read the Troubleshooting Guide and report your findings. That already helps.

**Comment by unicorn855** [comment]
After going through the troubleshooting steps, I realised that what I experienced can only be described as user error, as I was not in the habit of unlocking my database *before* starting Edge (on windows 11). Consequently, the the extension complained that it couldn't connect because the receiving end didn't exist (error in page.js:233 and before that in keepass.js:404) As long as I unlock before starting the browser, it connects fine 👍 and throws no error.

**Comment by unicorn855** [comment]
Indeed so, yes @dark-rider :-) I set the auto-locking option in KeePassXC, but for some reason, re-locking the database (and subsequently re-unlocking it) does not cause the connection to break as long as it was unlocked before Edge is started the first time after booting. Is that behaviour considered a bug? It is consistently working fine as long as I follow the flow described in my previous comment, so I am not sure.

**Comment by droidmonkey** [comment]
There is no requirement for the database to be unlocked prior to browser launch. You likely need to hit Refresh on the extension menu to reestablish a connection to keepassxc on browser launch. Database lock state is not relevant.

**Comment by unicorn855** [comment]
Well, yes that would also work, but then, why does the extension throw the errors I mentioned in my first reply, however it works fine (no errors) when I unlock before first launch? If this is indeed a bug, is there something I can do to help troubleshoot it beyond just reporting my findings of the troubleshooting page?

**Comment by unicorn855** [comment]
I am running KeePassXC 2.7.1 on Windows 11 (build 22622.290) and Edge version 103.0.1264.49 :-) sorry I completely forgot to mention that.

**Comment by unicorn855** [comment]
So, I have been checking the generated background page for the extension in Edge for a little while, and I am wondering if it's normal to continually get the "cannot send activated_tab message: Could not establish connection. Receiving end does not exist" error?
Further, I have done more testing about how the extension connects to a database, and it continues to connect fine the first time when the database is unlocked before browser launch, however, it fails to connect when the db is locked instead; after manually clicking "reload" it works fine. Is it currently expected to not automatically connect like this? How else can I offer more helpful input? The keepass_proxy.exe autostarts and keeps running fine in the background, btw. No other errors are thrown.

**Comment by unicorn855** [comment]
I don't know if this is worth opening a new issue, but perhaps it isn't, anyway, is there a reason why the extension is still on version 1.8.0 in the Edge Add-ons store, while its current (1.8.1) in chrome's extension store? Should I migrate to the version offered in the chrome store and if so, do I need to re-register all connected databases or can I somehow 'transfer' them to the chrome version (I still want to keep using Edge as a browser)?

**Comment by varjolintu** [comment]
@unicorn855 This is strange because my databases are 90% locked when I start Edge for any testing, and I have never had any problems. Are you able to debug the background scripts and see where the connect fails?

Microsoft has a very slow update queue in their Add-ons store. Maybe they're still doing some manual checks. Usually it takes 2-14 days.

**Comment by unicorn855** [comment]
> @unicorn855 This is strange because my databases are 90% locked when I start Edge for any testing, and I have never had any problems. Are you able to debug the background scripts and see where the connect fails?

I will figure out how to do that and report any findings :) 

> Microsoft has a very slow update queue in their Add-ons store. Maybe they're still doing some manual checks. Usually it takes 2-14 days.

I see, good to know, I'll just wait for a bit longer then 👍

**Comment by varjolintu** [comment]
Manual connection is mandatory if KeePassXC is not started before the browser, or is shut down in between.

**Comment by unicorn855** [comment]
Ah interesting! I suspect the issues I am experiencing might have to do with Edge's 'startup boost' option then, as that apparently starts up Edge with "minimal processes" (whatever MS means by that). This leads me to speculate that *some* components of the browser have started before KeepassXC has finished loading (which is also set to auto-start at boot time) completely, thus causing this behaviour. I'll turn startup boost off and test, then let you know if that resolved things :-)

**Comment by varjolintu** [comment]
@Romich777 Did the same solution worked for you?

**Comment by Romich777** [comment]
I removed all snap related soft and files, reinstalled Firefox and integration with KP became to work. Probably issue were with FF and snap but not with KP in my case

Sorry that cant to provide any logs/debug info
